### PR TITLE
CSV: standardize error handling

### PR DESCRIPTION
**Pull request by tsmacdonald**
All CSV-related errors will reliably have a `:message` key. The status code will be 422 if it's something we anticipated and 500 if it's not. There will no longer be a `:cause` key under any circumstance.

### CODE REVIEW

**Review by calherries — APPROVED** [review]
LGTM

File: src/metabase/api/card.clj [hunk 1]
```diff
@@ -1223,11 +1223,17 @@ saved later when it is ready."
...
-     :body   model-id}))
+  ;; parse-long returns nil with "root" as the collection ID, which is what we want anyway
+  (try
+    (let [model-id (:id (upload-csv! (parse-long (get raw-params "collection_id"))
+                                     (get-in raw-params ["file" :filename])
+                                     (get-in raw-params ["file" :tempfile])))]
+      {:status 200
+       :body   model-id})
+    (catch Throwable e
+      {:status (or (-> e ex-data :status-code)
```
Comment (qnkhuat) [inline src/metabase/api/card.clj:1234]: is a test needed?

**Comment by iethree** [comment]
A PR description would be helpful here for non-clojurians. Does this ensure that all errors end up in a `message` key?

**Comment by tsmacdonald** [comment]
@iethree sorry, posted it in Slack but not here.

> TL;DR is that you can depend on a `:message` key. The status code will be 422 if it's something I've anticipated and 500 if it's not. I've also done away with `:cause`

**Review by iethree — APPROVED** [review]
💖
